File: cf_sdk/src/main/java/com/example/cf_sdk/logic/sdk_module/creator/AccountApiCreator.java
package com.example.cf_sdk.logic.sdk_module.creator;

import com.example.cf_sdk.logic.changebankapi.network.api.AccountApi;
import com.example.cf_sdk.logic.changebankapi.network.api.ApiCreator;
import com.example.cf_sdk.logic.changebanksdk.log.Logger;

import retrofit2.Retrofit;

/**
 *
 * Network {@link AccountApi} creator.
 */

public class AccountApiCreator implements ApiCreator<AccountApi> {
    @Override
    public AccountApi create(Logger logger, Retrofit retrofit) {
        return retrofit.create(AccountApi.class);
    }
}

File: cf_sdk/src/main/java/com/example/cf_sdk/changebankapi/network/api/AccountApiCreator.kt
package com.example.cf_sdk.changebankapi.network.api

import com.example.cf_sdk.changebankapi.log.Logger
import retrofit2.Retrofit

/**
 * Created by victorojeda on 1/29/18.
 *
 * Network [AccountApi] creator.
 */
class AccountApiCreator : ApiCreator<AccountApi?> {
    override fun create(logger: Logger?, retrofit: Retrofit?): AccountApi {
        return retrofit!!.create(AccountApi::class.java)
    }
}